src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeEventInfo.cs

```diff
@@ -68,6 +68,13 @@ public override string ToString()
 
             return m_addMethod.GetParametersNoCopy()[0].ParameterType.FormatTypeName() + " " + Name;
         }
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
```

Comment:
Same as change to `ReferenceEquals` as https://github.com/dotnet/runtime/pull/73009/files#r935802561